gdal/port/cpl_json.cpp

```diff
@@ -0,0 +1,1217 @@
+/******************************************************************************
+ * Project:  Common Portability Library
+ * Purpose:  Function wrapper for libjson-c access.
+ * Author:   Dmitry Baryshnikov, dmitry.baryshnikov@nextgis.com
+ *
+ ******************************************************************************
+ * Copyright (c) 2016-2017 NextGIS, <info@nextgis.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "cpl_json.h"
+
+
+#include "cpl_error.h"
+#include "cpl_vsi.h"
+
+#include "json.h"
+
+#undef json_object_object_foreachC
+#define json_object_object_foreachC(obj,iter) \
```

Comment:
There's ogr/ogr_json_header.h that does that and include json.h with all warning suppressions

Reply:
This header file is in ogr folder. In current solution I try to hide everything dependent on json-c library and cpl_json.h has no dependencies. Also I try make port not dependent on ogr or very soft dependent. 

I think moving driver to this solution will make ogr_json_header.h unnecessary.

Reply:
My implied point was to suggest moving ogr/ogr_json_header.h as port/cpl_json_header.h. We do need the "#pragma GCC system_header" since there are a number of things in json.h and related headers that cause warnings with compilers

Reply:
Fixed